File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool Comp(int a, int b){
    return a>b;
}

vector<int> solution(int k, vector<int> score) {
    vector<int> answer;
    vector<int> rank;
    for(int i=0; i<score.size(); i++){
        if(rank.size() < k){
            rank.push_back(score[i]);
        }
        else{
            if(rank.back() < score[i]){
                rank.pop_back();
                rank.push_back(score[i]);
            }
        }
        
        sort(rank.begin(), rank.end(), Comp);
        answer.push_back(rank.back());
    }
    
    return answer;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    var answer = [];
    let rank = [];
     score.forEach(num => {
        rank.push(num);
        // 정렬하고 k개로 분할
        rank.sort((a,b) => b-a).splice(k);
         // 최솟값 넣기
        answer.push(Math.min.apply(null, rank));
    });
    
    return answer;
}